HAL/src/eoss3_hal_i2c.c

```diff
@@ -230,6 +237,7 @@ HAL_StatusTypeDef HAL_I2C_Write(UINT8_t ucDevAddress, UINT8_t ucAddress, UINT8_t
       	eI2CState = I2C_READY;
 	return HAL_ERROR;
     }
+    waitFFEReady(MAX_CYCLES_FFE);

```

Comment:
So how does the original code from qorc-sdk work, if they are not checking the FFE CSR value?

Reply:
When running it on the Quick Thing Plus board, it did not work with qorc-sdk samples either - I2C transmissions were failing. However, I will try and reflash the board with the original qorc-sdk code without my changes to confirm this and will share my findings. If indeed it is broken, I might also suggest this change to qorc-sdk.

without doing these checks however, the transmission failed every time on Zehyr as well. My suspicion are:
- this is broken on qorc-sdk as well, and if it is working then there is some delay introduced by other threads, which is not present on Zephyr (hence for stability, this wait is required)
- this is working on qorc-sdk and I misconfigured the I2C for qorc-sdk when testing it for QuickThing Plus with qorc-sdk. In this case, maybe the clock speed configured here https://github.com/zephyrproject-rtos/hal_quicklogic/pull/4/files/7f2916b388e26ee4bace4779da0ce5c2695aafca..e348ffd18778fc1aa212f830bd2b97b61a6cbcc4#diff-87abd87ab59fba2c37f505e3cb82a2aefdefdcd07126f4391195a971889dc29eR137 is incorrect, and if I reconfigure it properly, the issue will go away?

This particular issue is slightly beyond my knowledge of the underlying hardware, but I will try my best to give you a better answer, and test results